src/containers/viewAllArticles.js

```diff
@@ -1,50 +1,78 @@
 import React from 'react';
-import { withRouter } from 'react-router-dom';
+import { withRouter} from 'react-router-dom';
 import { connect } from 'react-redux';
 import * as PropTypes from 'prop-types';
 import styles from '../styles/articleStyles/viewArticle.scss';
 import ViewCard from '../components/articles/viewCard';
 import {fetchAllArticles} from '../actions/articleActions/articleActions';
 import {viewAllArticleActionCreator} from '../actions/articleActions/articleActionCreators';
+import Pagination from './pagination';
 
 class ViewAllArticles extends React.Component {
 
 	constructor(props) {
 		super(props);
 		this.state = {
-			articles: this.props.articles.results
+			articles: this.props.articles,
+			pageCount: 0,
+			page: 1,
+			limit: 10,
+			offset: 0
 		};
 	}
 
 	componentWillMount() {
-		if (!this.state.articles || this.state.articles.length < 1) {
-			this.props.dispatch(fetchAllArticles());
-		} else { this.props.dispatch(fetchAllArticles(false)); }
+		if (!this.state.articles.results || this.state.articles.count < 1) {
+			this.loadAllArticles(true);
+		} else { this.loadAllArticles(false); }
 
 		this.props.dispatch(viewAllArticleActionCreator());
 	}
 
 	componentDidMount() {
 		this.timer = setInterval(this.loadAllArticles, 10000);
+		this.timer2 = setInterval(this.loadViewAll, 1000);
 	}
 
 	// noinspection JSUnusedLocalSymbols
 	componentWillReceiveProps(nextProps, nextContext) {
-		this.setState({articles: nextProps.articles.results});
+		this.setState(prevState => ({
+			articles: nextProps.articles,
```

Comment:
here

Reply:
I have fixed this to use destructuring assignment.